microsetta_public_api/api/taxonomy.py

```diff
@@ -45,6 +46,59 @@ def summarize_group(body, resource):
     return _summarize_group(sample_ids, resource, taxonomy_repo)
 
 
+LEVEL_MAP = {
+    'domain': 0,
+    'phylum': 1,
+    'class': 2,
+    'order': 3,
+    'family': 4,
+    'genus': 5,
+    'species': 6
+}
+
+
+def group_counts(body, dataset, resource, level):
+    taxonomy_repo = _get_taxonomy_repo(dataset)
+    sample_ids = body['sample_ids']
+
+    error_response = _check_resource_and_missing_ids(taxonomy_repo,
+                                                     sample_ids, resource)
+    if error_response:
+        return error_response
+
+    if sample_ids == []:
+        sample_ids = None
+
+    level = LEVEL_MAP.get(level)
+    if level is None:
+        raise UnknownLevel("Level must be one of: %s" % list(LEVEL_MAP.keys()))
+
+    taxonomy_ = taxonomy_repo.model(resource)
+
+    counts = taxonomy_.get_counts(level, sample_ids)
+    response = jsonify(counts)
+    return response, 200
```

Comment:
It looks like the entire logic of the function after extracting re-structuring `sample_ids` is repeated. Possible to extract to a function?

Reply:
👍